cms/djangoapps/contentstore/asset_storage_handlers.py

```diff
@@ -0,0 +1,621 @@
+    # keep things a bit more consistent
+    return {
+        'upload_file': upload_file,
+        'filename': upload_file.name,
+        'mime_type': upload_file.content_type,
+        'upload_file_size': get_file_size(upload_file)
+    }
+
+
+def get_file_size(upload_file):
+    """returns the size of the uploaded file"""
+    # can be used for mocking test file sizes.
+    return upload_file.size
+
+
+def _check_file_size_is_too_large(file_metadata):
+    """verifies whether file size is greater than allowed file size"""
+    upload_file_size = file_metadata['upload_file_size']
+    maximum_file_size_in_megabytes = settings.MAX_ASSET_UPLOAD_FILE_SIZE_IN_MB
+    maximum_file_size_in_bytes = maximum_file_size_in_megabytes * 1000 ** 2
+
+    return upload_file_size > maximum_file_size_in_bytes
+
+
+def _get_file_too_large_error_message(filename):
+    """returns formatted error message for large files"""
+
+    return _(
+        'File {filename} exceeds maximum size of '
+        '{maximum_size_in_megabytes} MB.'
+    ).format(
+        filename=filename,
+        maximum_size_in_megabytes=settings.MAX_ASSET_UPLOAD_FILE_SIZE_IN_MB,
+    )
+
+
+def _get_file_content_and_path(file_metadata, course_key):
+    """returns contents of the uploaded file and path for temporary uploaded file"""
+    content_location = StaticContent.compute_location(course_key, file_metadata['filename'])
+    upload_file = file_metadata['upload_file']
+
+    file_can_be_chunked = upload_file.multiple_chunks()
+
+    static_content_partial = partial(StaticContent, content_location, file_metadata['filename'],
+                                     file_metadata['mime_type'])
+
+    if file_can_be_chunked:
+        content = static_content_partial(upload_file.chunks())
+        temporary_file_path = upload_file.temporary_file_path()
+    else:
+        content = static_content_partial(upload_file.read())
+        temporary_file_path = None
+    return content, temporary_file_path
+
+
+def _check_thumbnail_uploaded(thumbnail_content):
+    """returns whether thumbnail is None"""
+    return thumbnail_content is not None
+
+
+def _get_thumbnail_asset_key(asset, course_key):
+    """returns thumbnail asset key"""
+    # note, due to the schema change we may not have a 'thumbnail_location' in the result set
+    thumbnail_location = asset.get('thumbnail_location', None)
+    thumbnail_asset_key = None
+
+    if thumbnail_location:
+        thumbnail_path = thumbnail_location[4]
+        thumbnail_asset_key = course_key.make_asset_key('thumbnail', thumbnail_path)
+    return thumbnail_asset_key
+
+
+# TODO: this method needs improvement. These view decorators should be at the top in an actual view method,
+#  but this is just a method called by the asset_handler. The asset_handler used by the public studio content API
+# just ignores all of this stuff.
+@require_http_methods(('DELETE', 'POST', 'PUT'))
+@login_required
+@ensure_csrf_cookie
```

Comment:
@kdmccormick I'd like to, but it's risky since I don't know that for sure, and I consider it out of scope for this. when I don't have a very deep functionality of something I work on, I restrict my refactoring to known patterns that don't change functionality (like extract method pattern, and others). I have not written the offending code and this is a cleanup I consider out of scope here, because I aim to keep risk to a minimum. We can open a maintenance issue for this that involves sufficiently testing the removal, and put it in our backlog, if you insist on it.

Reply:
I see. It makes sense that you'd want to avoid any substantive change in your refactoring. Up to you whether you make the maintenance ticket.